ui/vic-ui-h5c/vic/src/vic-webapp/src/app/shared/utils/regex.ts

```diff
@@ -14,3 +14,13 @@
  limitations under the License.
 */
 export const camelCasePattern = new RegExp(/([a-z])([A-Z])/g);
+export const kebabCasePattern = new RegExp(/^[\w-]+$/);
```

Comment:
Just make sure these all have been thoroughly validated (using online regex tester) with all the different input variations

Reply:
This is allowing any word and/or hyphens. In [Wizard Fields Validations](https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=~cfalcone&title=Create+VCH+Wizard), Alister commented: "Have to check this pattern. I suspect an underscore is also allowed". This pattern is used for VCH Name, Container Network Label, and Docker Volume Name fields. What I found is that vic-machine create command outputs an error with the following message when an unsupported char is entered: "--name contains unsupported characters: ?my@name Allowed characters are alphanumeric, space and symbols - _ ( )". Also in vic-machine create command sources they are disallowing name fields values that match the regex: /%|&|\*|\$|#|@|!|\\|/|:|\?|"|<|>|;|'|\|/. So, maybe we should use /^[^%|&|\*|\$|#|@|!|\\|\/|:|\?|"|<|>|;|'|\|]+$/ regex for this kind of fields, what do you think?

Reply:
It sounds good to me as long as we stick with the standards the vic-machine CLI uses.

Reply:
Agreed